e2e/spec/bugReporting/bugReportingPhase.e2e-spec.ts

```diff
@@ -0,0 +1,44 @@
+import { BugReportingPhase } from '../../page-objects/bugReportingPhase.po';
+import { PhaseDescription } from '../../../src/app/core/services/phase.service';
+import { Phase } from '../../../src/app/core/models/phase.model';
+import { LoginPage } from '../../page-objects/login.po';
+import { Header } from '../../page-objects/header.po';
+
+describe('CATcher\'s Bug Reporting Page', () => {
+  let bugReportingPhase: BugReportingPhase;
+  let loginPage: LoginPage;
+  let header: Header;
+
+  beforeEach(() => {
+    loginPage = new LoginPage();
+    header = new Header();
+    bugReportingPhase = new BugReportingPhase();
+    loginPage.navigateToRoot();
+  });
+
+  it(`displays "${PhaseDescription[Phase.phaseBugReporting]}" in header bar`, async () => {
+    await loginPage.bypassAuthentication();
+    expect(await bugReportingPhase.getPhaseDescription()).toContain(PhaseDescription[Phase.phaseBugReporting]);
+  });
+
+  it('creates new bug report', async () => {
+    const testIssueCreationTitle = 'Test Issue Creation Title';
+    const testIssueCreationDescription = 'Test Issue Creation Text';
+
+    await loginPage.bypassAuthentication();
+    const issueCount: number = await bugReportingPhase.getNumberOfBugReports();
+
+    await bugReportingPhase.accessNewBugReportingPage()
+      .then(() => bugReportingPhase.enterNewIssueTitle(testIssueCreationTitle))
+      .then(() => bugReportingPhase.enterNewBugReportDescription(testIssueCreationDescription))
+      .then(() => bugReportingPhase.selectSeverityDropdown())
+      .then(() => bugReportingPhase.selectDropDownOption())
+      .then(() => bugReportingPhase.selectBugTypeDropdown())
+      .then(() => bugReportingPhase.selectDropDownOption())
+      .then(() => bugReportingPhase.submitBugReport())
+      .then(() => header.clickBackButton());
```

Comment:
Was trying to find a clean way to "chain" these actions as it felt weird having to use the `await` keyword for each step instead of having it flow together.

I thought of converting the original `Promise` to an `Observable` then present each sub-step as a custom pipe operator but that would make the individual functions less flexible to use when writing future testcases as we'd have to decide which functions would more appropriately fit as operators and which as observables.

Any suggestions/ideas? @CATcher-org/2021-devs